components/AppHeader.vue

```diff
@@ -0,0 +1,87 @@
+<script setup lang="ts">
+import { CIcon } from '@coreui/icons-vue'
+import throttle from 'lodash.throttle'
+import { useRoute } from 'vue-router'
+
+const emit = defineEmits(['toggleSidebar'])
+
+const route = useRoute()
+
+const showBreadcrumbs = computed(() => route.meta.hideBreadcrumbs !== true)
+
+function toggleSidebar() {
+  emit('toggleSidebar')
+}
+
+// We apply a shadow to the header when the position is scrolled down
+const isScrolled = ref(false)
+const handleScroll = throttle(() => {
+  isScrolled.value = document.documentElement.scrollTop > 0
+}, 100)
+
+onMounted(() => {
+  window.addEventListener('scroll', handleScroll)
+})
+
+onBeforeUnmount(() => {
+  window.removeEventListener('scroll', handleScroll)
+})
+</script>
+
+<template>
```

Comment:
I've always preferred putting the template above the script, as it feels more natural to see your static markup before the dynamic operations which work upon it. But again, this is quite subjectve!

Reply:
I agree actually, so I'm now overriding the lint/format rule that caused this.